pkg/payment_request/service_param_value_lookups/service_param_value_lookups.go

```
@@ -76,10 +76,10 @@ func ServiceParamLookupInitialize(
 			return nil, services.NewNotFoundError(uuid.Nil, "looking for PickupAddressID")
 		}
 		pickupAddress = *mtoShipment.PickupAddress
-		if mtoShipment.DestinationAddressID == nil {
-			return nil, services.NewNotFoundError(uuid.Nil, "looking for DestinationAddressID")
+		if mtoShipment.DestinationAddressID != nil {
```

Before the refactor, service items that required destination address would throw an error so this changes that. Can you throw the error from inside the specific lookup that does require destination to be present? Also, I'm wondering in real world will we ever not have the dest address when a payment request is made? Second I'm wondering if we need to do this with pickup address too

---

> Can you throw the error from inside the specific lookup that does require destination to be present?

Yeah I'll try that. @pearl-truss and I ran into this being an issue.

> Also, I'm wondering in real world will we ever not have the dest address when a payment request is made?

Hmm, not sure.

> Second I'm wondering if we need to do this with pickup address too

Yeah, I guess if there may not be a destination, it might also be missing a pickup?

---

@gidjin I updated the PR to restore the error upstream not run the lookups that require destination address if it's a `DPK` service item. If I can get your 👀 on verifying that you can create a `DPK` and a `DUPK` for good measure, that'd be awesome :)